Abandon Session Mode «Load-Dependent Session Settings»

mBeym opened the pull request:
Removes everything related to the load-dependent session settings

Feature Wiki Page: https://docu.ilias.de/goto_docu_wiki_wpage_8263_1357.html

* @Uwe-Kohnle: We had to change one line in the SCORM components
* @kergomard: We removed the "Session Mode" setting from the user component

A quick review is highly appreciated.

mBeym committed:
Remove translations

mBeym committed:
Remove Load-Dependent Session Settings

mjansenDatabay commented on components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php:
```diff
@@ -75,4 +75,42 @@ public function step_4(): void
         $this->db->manipulate("DELETE FROM settings WHERE keyword LIKE 'char_selector%'");
     }
+
+    public function step_5(): void
```
Please move this to an own database objective in the `Authentication` component. You can name it `AbandonLoadDependantSessionDatabaseUpdateObjective`.

mjansenDatabay commented on components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php:
```diff
@@ -75,4 +75,42 @@ public function step_4(): void
+        }
+    }
+
+    public function step_6(): void
+    {
+
+    }
+
+    public function step_7(): void
+    {
+
+    }
```
Why do we need these empty steps?

mjansenDatabay commented on components/ILIAS/Authentication/classes/class.ilSession.php:
```diff
@@ -353,58 +334,28 @@ public static function _duplicate(string $a_session_id): string
      *
      * Returns the expiration timestamp in seconds
      *
-     * @param	boolean	If passed, the value for fixed session is returned
```
We should cleanup this comment in general.

mjansenDatabay commented on components/ILIAS/Authentication/classes/class.ilSession.php:
```diff
@@ -353,58 +334,28 @@ public static function _duplicate(string $a_session_id): string
      *
      * Returns the idle time in seconds
      *
-     * @param	boolean	If passed, the value for fixed session is returned
```
We should cleanup this comment in general.

mBeym replied:
just remove the `@return` or the entire comment?

mjansenDatabay commented on components/ILIAS/Authentication/classes/class.ilSessionControl.php:
```diff
@@ -27,10 +27,7 @@ class ilSessionControl
      * been defined in setup or administration yet
      */
-    public const DEFAULT_MAX_COUNT = 0;
     public const DEFAULT_MIN_IDLE = 15;
```
Do we still need the "Min Idle Time" setting?

mBeym replied:
it's still being used in ilSCORM13PlayerGUI.

I could move that constant into the scorm class directly or replace it with hard coded values though.

mjansenDatabay commented on components/ILIAS/Authentication/templates/default/tpl.session_statistics_center.html:
```diff
@@ -12,6 +12,4 @@ <h3>{TIMEFRAME}</h3>
 			<li>{CAPTION_CLOSED_DETAILS}: {VALUE_CLOSED_DETAILS}</li>
 			<!-- END closed_details -->
 		</ul>		
-		<p>{CAPTION_MAXED_OUT_COUNTER}: {VALUE_MAXED_OUT_COUNTER}</p>
```
Maybe we should keep this, so the "Session Statistics" can work with the old values.

mBeym replied:
I couldn't find any code line where this variable is still set.

mjansenDatabay commented on components/ILIAS/Authentication/classes/class.ilSession.php:
```diff
@@ -52,7 +34,6 @@ class ilSession
     public const SESSION_CLOSE_EXPIRE = 2;  // has expired
     public const SESSION_CLOSE_FIRST = 3;  // kicked by session control (first abidencer)
     public const SESSION_CLOSE_IDLE = 4;  // kickey by session control (ilde time)
-    public const SESSION_CLOSE_LIMIT = 5;  // kicked by session control (limit reached)
```
Why do we delete this but keep `SESSION_CLOSE_FIRST`?

mjansenDatabay reviewed (changes requested):
Thx @mBeym,

- [x] Please revert all language file related changes except those in `ilias_de.lang` and `ilias_en.lang`.
- [x] Please move the database update steps to the `Authentication` component.
- [x] In my opinion, we could remove more constants in `ilSession` but maybe have to migrate or even delete data/drop table columns from the session statistics database tables. I suggest activating the feature in “Fixed session duration” mode and checking which database fields can be dispensed with.

Best regards,
Michael

mjansenDatabay commented:
@mBeym Please resolve the conflicts

mBeym committed:
Remove comments

mBeym commented:
> @mBeym Please resolve the conflicts

done

kergomard reviewed (approved):
Thank you very much for the PR @mBeym and @mjansenDatabay !

I'm fine with the changes in `ilObjUserFolderGUI`.

If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a  checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).

And a very small suggestion:
Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.

Thanks again and best,
@kergomard

mjansenDatabay commented:
> Thank you very much for the PR @mBeym and @mjansenDatabay !
> 
> I'm fine with the changes in `ilObjUserFolderGUI`.
> 
> If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).
> 
> And a very small suggestion: Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.
> 
> Thanks again and best, @kergomard

Thx @kergomard , the old settings should be deleted here: https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-8b1ca2c61f7661b03b2012f2337b9baba324b4eb4ebb25c5128b3ea40f55d72aR37

@mBeym Can you please rename the variable?

mBeym committed:
Give inputs meaningful names

kergomard commented:
Thank you very much for the pointer @mjansenDatabay ! Sorry for not having read carefully enough (one should not collapse the files overview and forget one has done so)!

mBeym commented:
> > Thank you very much for the PR @mBeym and @mjansenDatabay !
> > I'm fine with the changes in `ilObjUserFolderGUI`.
> > If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).
> > And a very small suggestion: Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.
> > Thanks again and best, @kergomard
> 
> Thx @kergomard , the old settings should be deleted here: https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-8b1ca2c61f7661b03b2012f2337b9baba324b4eb4ebb25c5128b3ea40f55d72aR37
> 
> @mBeym Can you please rename the variable?

Done. Renamed to **$session_reminder**.

@kergomard I also renamed the **$ti** ilNonEditableValueGUI variable to **$session_config** if that's ok for you?

mjansenDatabay commented:
Hi @pascalseeland ,

I would like to merge this within the next week. Any feedback is highly appreciated.

Best regards,
Michael

mBeym committed:
Fix code-style

mBeym committed:
Fix code-style
